dio/lib/src/options.dart

```diff
@@ -100,7 +100,7 @@ class BaseOptions extends _RequestConfig with OptionsMixin {
     ResponseDecoder? responseDecoder,
     ListFormat? listFormat,
   })  : assert(connectTimeout == null || !connectTimeout.isNegative),
-        assert(baseUrl.isEmpty || Uri.parse(baseUrl).host.isNotEmpty),
+        assert(baseUrl.isEmpty || Uri.parse(baseUrl).host.isEmpty),
```

Comment:
Maybe you should add a test case for it.

Reply:
Can you change the existing test here: https://github.com/cfug/dio/blob/main/dio/test/options_test.dart#L380 to test the actual assert when constructing the `BaseOptions` please. Currently we are just testing a Dart SDK function which doesn't help us.